addons/binding/org.openhab.binding.tplinksmarthome/src/main/java/org/openhab/binding/tplinksmarthome/internal/handler/SmartHomeHandler.java

```diff
@@ -105,15 +120,64 @@ Connection createConnection(TPLinkSmartHomeConfiguration config) {
         return new Connection(config.ipAddress);
     }
 
-    private DeviceState refreshCache() {
+    private Optional<DeviceState> refreshCache() {
```

Comment:
Does this make much sense now you have to handle and optional, should you just make this method return a Nullable DeviceState?

Reply:
Yes it does make sense. It was the only simple way I could think of to get this to work without getting compiler errors. The problem is I added `@NonNullByDefault` (silly me :smile:) And this causes a problem with the `ExpiringCache`. The problem is the method `getValue` on that class is `Nullable`. But the only way to get a null value is if the `Supplier` would return a null. However because the `@NonNullByDefault` in the class enforces the non nullable on the Supplier. This would all not be a problem if my implementation wasn't based on a `getValue` that does return null in case no value could be given. So I have a value that can be null, but the Supplier doesn't accept the value to be null. To fix this the method should not return null anymore. One way is to introduce an empty object that is returned when no value is present, but that is basically the whole reason to have an Optional. Hence the Optional was introduced.

Reply:
Wouldn’t it make more sense to mark it Nullable? Because it shall also return null if the key is not available.

Reply:
I added a PR to your repo and one to ESH:

https://github.com/Hilbrand/openhab2-addons/pull/1
https://github.com/eclipse/smarthome/pull/6485